src/app/pages/products-list/card/card.component.ts

```diff
@@ -1,8 +1,43 @@
-import {Component} from '@angular/core';
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {Product} from 'src/app/shared/products/product.interface';
+
+const MAX_RATING = 5;
+
+interface RatingArr {
+    active: boolean;
+}
 
 @Component({
     selector: 'app-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.css'],
 })
-export class CardComponent {}
+export class CardComponent {
+    @Input() product: Product | null = null;
+
+    @Output() buyProduct: EventEmitter<Product | null> = new EventEmitter();
+
+    get cardImageUrl() {
+        return this.product?.images[0].url;
+    }
+
+    getRating(ratingNum: number): RatingArr[] {
+        const ratingArr: RatingArr[] = [];
+
+        for (let i = 1; i <= MAX_RATING; i++) {
+            const ratingObj = {
+                active: i <= ratingNum,
+            };
+
+            ratingArr.push(ratingObj);
+        }
+
+        return ratingArr;
```

Comment:
Я бы предложил решить это чуть иначе: пока что не видится особого смысла в держании объектов в массиве, я предложил бы держать булевые значения; а собирал бы массив применяя методы массивов:
```suggestion
        return Array.from({length: MAX_RATING}, (_value, index) => index + 1 <= ratingNum)
```

А в шаблоне использовали бы это так:
```html
<mat-icon *ngFor="let isActive of getRating(product.rating)" [class.star-active]="isActive">star</mat-icon>
```

Reply:
исправлено